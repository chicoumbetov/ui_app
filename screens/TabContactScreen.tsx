import * as React from 'react';
import { StyleSheet } from 'react-native';

import { Layout, Text } from '@ui-kitten/components';

export default function TabContactScreen() {
  return (
    <Layout style={styles.container}>
      <Text>Tab Contact</Text>

    </Layout>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'rgba(246, 246, 246, 0.5)',
  },
});
