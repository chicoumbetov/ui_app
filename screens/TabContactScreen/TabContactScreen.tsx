/**
 * Contact Screen to access in Contact Page and visualize
 *
 * @author: Shynggys UMBETOV
 */

import * as React from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';

import { Layout } from '@ui-kitten/components';
import { createStackNavigator } from '@react-navigation/stack';
import { AntDesign } from '@expo/vector-icons';
import Contact from './Components/Contact';
import HeaderLeftOpenDrawerNavigation from '../../navigation/HeaderLeftOpenDrawerNavigation';
import HeaderRightOpenDrawerNavigation from '../../navigation/HeaderRightOpenDrawerNavigation';
import Contact2 from './Components/Contact2';

const Stack = createStackNavigator();

const TabContactScreen = ({ navigation }) => (
  <Layout style={styles.container}>
    <Stack.Navigator
      initialRouteName="Contact"
    >
      <Stack.Screen
        name="Contact"
        options={{
          headerTitle: false,
          headerLeftContainerStyle: {
            paddingBottom: 5,
          },
          headerRightContainerStyle: {
            marginBottom: 5,
          },
          headerShown: true,
          headerLeft: () => (
            <HeaderLeftOpenDrawerNavigation navigation={navigation} />
          ),
          headerRight: () => (
            <HeaderRightOpenDrawerNavigation navigation={navigation} />
          ),
        }}
        component={Contact}
      />
      <Stack.Screen
        name="Contact2"
        options={{
          headerTitle: false,
          headerLeftContainerStyle: {
            paddingBottom: 5,
          },
          headerRightContainerStyle: {
            marginBottom: 5,
          },
          headerShown: true,
          headerLeft: () => (
            <TouchableOpacity onPress={() => { navigation.navigate('Contact'); }}>
              <AntDesign name="arrowleft" size={30} style={{ color: '#b5b5b5', marginLeft: 20 }} />
            </TouchableOpacity>
          ),
          headerRight: () => (
            <HeaderLeftOpenDrawerNavigation navigation={navigation} />
          ),
        }}
        component={Contact2}
      />
    </Stack.Navigator>
  </Layout>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'rgba(246, 246, 246, 0.5)',
  },
});

export default TabContactScreen;
