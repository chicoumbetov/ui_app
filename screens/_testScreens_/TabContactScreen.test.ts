import renderer from 'react-test-renderer';

import TabContactScreen from '../TabContactScreen';

it('renders TabContactScreen', () => {
  expect(TabContactScreen);
});

describe('<TabContactScreen />', () => {
  it('renders correctly across screens', () => {
    const tree = renderer
      .create('<TabContactScreen />').toJSON();
    expect(tree).toMatchSnapshot();
  });
});
