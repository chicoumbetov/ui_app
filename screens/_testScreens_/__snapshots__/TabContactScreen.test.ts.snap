// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabContactScreen /> renders correctly across screens 1`] = `"<TabContactScreen />"`;
